cmd/manila-csi-plugin/main.go

```diff
@@ -106,9 +124,6 @@ func main() {
 	cmd.PersistentFlags().StringVar(&driverName, "drivername", "manila.csi.openstack.org", "name of the driver")
 
 	cmd.PersistentFlags().StringVar(&nodeID, "nodeid", "", "this node's ID")
-	if err := cmd.MarkPersistentFlagRequired("nodeid"); err != nil {
-		klog.Fatalf("Unable to mark flag nodeid to be required: %v", err)
-	}
```

Comment:
Should we still do it when `provideNodeService` is true?

Reply:
There's a new validation in [Driver.SetupNodeService()](https://github.com/kubernetes/cloud-provider-openstack/pull/2316/files#diff-9aa99c6124af691916f6072fbc83e46e895d21a9df845f016ea5ebe38d587bdaR180) ensuring that the `nodeid` is not empty if required.

Marking it as required sounds wrong if `provideNodeService` is `false`, marking it only required if `provideNodeService` is `true` seems to be weird. The description of the `PersistentFlag` may be extended to note that it is required for the node service?

Reply:
Yeah, true, we could update the description, otherwise looks good. Please update that, I'll do `/approve` anyway.